File: app/src/main/java/com/scatl/uestcbbs/module/message/adapter/AtMeMsgAdapter.java
package com.scatl.uestcbbs.module.message.adapter;

import android.view.View;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.scatl.uestcbbs.R;
import com.scatl.uestcbbs.entity.AtMsgBean;
import com.scatl.uestcbbs.helper.glidehelper.GlideLoader4Common;
import com.scatl.uestcbbs.util.TimeUtil;

/**
 * author: sca_tl
 * description:
 * date: 2020/1/29 13:08
 */
public class AtMeMsgAdapter extends BaseQuickAdapter<AtMsgBean.BodyBean.DataBean, BaseViewHolder> {

    public AtMeMsgAdapter(int layoutResId) {
        super(layoutResId);
    }

    @Override
    protected void convert(BaseViewHolder helper, AtMsgBean.BodyBean.DataBean item) {
        helper.setText(R.id.user_name, item.user_name)
                .setText(R.id.reply_content, item.reply_content.replaceAll("\r\n", ""))
                .setText(R.id.board_name, "来自板块:" + item.board_name)
                .setText(R.id.subject_title, item.topic_subject)
                .setText(R.id.subject_content, item.topic_content.trim())
                .setText(R.id.reply_date,
                        TimeUtil.formatTime(item.replied_date, R.string.post_time1, mContext))
                .addOnClickListener(R.id.user_icon)
                .addOnClickListener(R.id.board_name);

        GlideLoader4Common.simpleLoad(mContext, item.icon, helper.getView(R.id.user_icon));
    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/message/adapter/AtMeMsgAdapter.kt
package com.scatl.uestcbbs.module.message.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import com.scatl.uestcbbs.databinding.ItemAtMeMsgBinding
import com.scatl.uestcbbs.entity.AtMsgBean
import com.scatl.uestcbbs.helper.PreloadAdapter
import com.scatl.uestcbbs.helper.ViewBindingHolder
import com.scatl.uestcbbs.util.TimeUtil
import com.scatl.uestcbbs.R
import com.scatl.uestcbbs.util.load

/**
 * Created by sca_tl at 2023/6/12 14:14
 */
class AtMeMsgAdapter: PreloadAdapter<AtMsgBean.BodyBean.DataBean, ItemAtMeMsgBinding>() {

    override fun getViewBinding(parent: ViewGroup): ItemAtMeMsgBinding {
        return ItemAtMeMsgBinding.inflate(LayoutInflater.from(context), parent, false)
    }

    override fun onBindViewHolder(holder: ViewBindingHolder<ItemAtMeMsgBinding>, position: Int, item: AtMsgBean.BodyBean.DataBean?) {
        super.onBindViewHolder(holder, position, item)
        if (item == null) {
            return
        }

        holder.binding.userName.text = item.user_name
        holder.binding.replyContent.text = item.reply_content.replace("\r\n".toRegex(), "")
        holder.binding.boardName.text = "来自板块:${item.board_name}"
        holder.binding.subjectTitle.text = item.topic_subject
        holder.binding.subjectContent.text = item.topic_content.trim()
        holder.binding.replyDate.text = TimeUtil.formatTime(item.replied_date, R.string.post_time1, context)
        holder.binding.userIcon.load(item.icon)
    }
}